packages/transit-vehicle-overlay/src/utils.js

```diff
@@ -0,0 +1,298 @@
+  query = query || "routes/all";
+  let url = buildUrl(baseUrl, query);
+
+  // append date to end of url (caching, etc...)
+  const d = Date.now();
+  url = url.indexOf("?") ? `${url}?` : `${url}&`;
+  url = `${url}__time__=${d}`;
+
+  fetch(url)
+    .then(res => {
+      if (!res.ok) {
+        console.log(res.statusText);
+        throw Error(res.statusText);
+      }
+      return res;
+    })
+    .then(res => {
+      const retVal = res.json();
+      return retVal;
+    })
+    .then(vehicleList => {
+      if (vehicleList && vehicleList.length > 0) {
+        // step 1: set the vehicle list (triggers vehicle points redraw)
+        console.log(`updating state with ${vehicleList.length} vehicles`);
+        setData(vehicleList, trackedId);
+      } else {
+        console.log("get vehicle data is suspect");
+      }
+    })
+    .catch(error => {
+      console.log(`VEH fetch() error: ${error} for ${url}`);
+    });
+}
+
+/**
+ * fetch line geometry for a given pattern
+ *
+ * https://newplanner.trimet.org/ws/ti/v0/index/patterns/TRIMET:433758/geometry/geojson
+ * https://newplanner.trimet.org/ws/ti/v0/index/patterns/{agency}:{pattern}/geometry/geojson
+ *
+ * @param setPatternData
+ * @param patternId
+ * @param tiUrl
+ * @param geojson
+ */
+export function fetchVehiclePattern(
+  setPatternData,
+  patternId,
+  tiUrl,
+  geojson = "/geojson"
+) {
+  if (!tiUrl) tiUrl = "https://newplanner.trimet.org/ws/ti/v0/index";
+  const url = `${tiUrl}/patterns/${patternId}/geometry${geojson}`;
+
+  console.log(`Calling GEO URL: ${url}`);
+  fetch(url)
+    .then(res => {
+      const retVal = res.json();
+      return retVal;
+    })
+    .then(json => {
+      if (geojson.indexOf("geojson") >= 0) {
+        json = reverseGeojsonPointsInGeom(json);
+      } else {
+        json = decodePolyline(json);
+      }
+      setPatternData(patternId, json);
+    })
+    .catch(error => {
+      console.log(`VEH GEOMETRY fetch() error: ${error}`);
+    });
+}
+
+/**
+ * get refresh values (default 10 seconds), and convert from secs to millisecs
+ */
+export function checkRefreshInteval(inverval, defInterval = 10000) {
+  let retVal = defInterval;
+  if (inverval) {
+    let r = inverval;
+    if (r > 0 && r <= 100) r *= 1000;
+    if (r >= 1000 && r < 100000) retVal = r;
+    else retVal = defInterval;
+  }
+  return retVal;
+}
+
+/**
+ * will build up a gtfsdb url for rt vehicles
+ * certain rules exist around the various filters
+ * url:
+ *   https://maps.trimet.org/gtfs/rt/vehicles/routes/all
+ */
+export function makeWsUrl(
+  url,
+  routes = null,
+  blocks = null,
+  trips = null,
+  stops = null
+) {
+  let filter = "";
+  if (routes) filter = `/routes/${routes}`;
+  else filter = "/routes/all";
+  if (blocks) filter = `/blocks/${blocks}`;
+  if (trips) filter = `/trips/${trips}`;
+  if (stops) filter = `/stops/${stops}`;
+
+  const retVal = url + filter;
+  return retVal;
+}
+
+export function formatTime(seconds) {
```

Comment:
Could core-utils/src/time.js#formatDuration be modified to satisfy this use case?